### PR TITLE
{chem}[fosscuda/2018b] Amber v18

### PR DESCRIPTION
**Pull request by akesandgren**
(created using `eb --new-pr`)
Amber 18.8 AmberTools 18.10

### CODE REVIEW

File: easybuild/easyconfigs/a/Amber/Amber-18-fosscuda-2018b-AmberTools-18-patchlevel-10-8.eb [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    '6b09a6548fa18127245e05c79bdca143adf31db349349cb39bfdb8c4f60772a7',  # Amber-18_fix_missing_build_target.patch
+    'a4c12ad39088ce6e9e7bad39e8d9a736989e6ae4a380c70535a5451eb3060903',  # Amber-18_dont_use_ipo.patch
+    '672c7667c899c91d002a4b095f1ac1691f77e7b4095414084f1d4cdc7c860d79',  # Amber-18_test_cuda.patch
+]
+
+dependencies = [
+    ('netCDF', '4.6.1'),
+    ('netCDF-Fortran', '4.4.4'),
+    ('Python', '2.7.15'),
+    ('Boost', '1.68.0', '-Python-%(pyver)s'),
```
Comment (boegel) [inline easybuild/easyconfigs/a/Amber/Amber-18-fosscuda-2018b-AmberTools-18-patchlevel-10-8.eb]: @akesandgren Please use `Boost.Python` as a dependency instead, if you need a `Boost` with Python bindings

Comment (akesandgren) [reply easybuild/easyconfigs/a/Amber/Amber-18-fosscuda-2018b-AmberTools-18-patchlevel-10-8.eb]: Didn't know it existed :-) Going blind on all eyes apparently.

Comment (boegel) [reply easybuild/easyconfigs/a/Amber/Amber-18-fosscuda-2018b-AmberTools-18-patchlevel-10-8.eb]: It's fairly recent, see https://github.com/easybuilders/easybuild-easyconfigs/pull/6763

Comment (akesandgren) [reply easybuild/easyconfigs/a/Amber/Amber-18-fosscuda-2018b-AmberTools-18-patchlevel-10-8.eb]: Then i should also drop down to 1.67 since that's the only existing Boost for a 2018b toolchain at the moment or?

Comment (boegel) [reply easybuild/easyconfigs/a/Amber/Amber-18-fosscuda-2018b-AmberTools-18-patchlevel-10-8.eb]: @akesandgren Indeed, for the `2018b` toolchain generation, we're sticking to Boost 1.67.0, to ensure all easyconfigs using a `2018b` toolchain that depend on `Boost` are compatible with each other...

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 8 out of 8 (8 easyconfigs in this PR)
b-an03.hpc2n.umu.se - Linux ubuntu 16.04, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 2.7.12
See https://gist.github.com/1b9c1bb6a4bc12cbc51f0d57661f2b29 for a full test report.

File: easybuild/easyconfigs/a/Amber/Amber-18-fosscuda-2018b-AmberTools-18-patchlevel-10-8.eb (hunk 1 above)
Comment (akesandgren) [reply easybuild/easyconfigs/a/Amber/Amber-18-fosscuda-2018b-AmberTools-18-patchlevel-10-8.eb]: Changes done.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 7 out of 7 (6 easyconfigs in this PR)
node2708.swalot.os - Linux centos linux 7.5.1804, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/ed2d5141db85371a3a83f92f6cf4dcce for a full test report.

**Comment by akesandgren** [comment]
Sigh.... some days one shouldn't try to work....

Better now?

**Comment by boegel** [comment]
OK for me now, but I can't test this since I don't have access to the `Amber` 18 sources... Maybe @vanzod and/or @migueldiascosta do though?

**Comment by akesandgren** [comment]
@vanzod @migueldiascosta Testing? and merging?

**Comment by akesandgren** [comment]
@migueldiascosta ping them again?

**Comment by branfosj** [comment]
Not sure whether to put this feedback here or in #7102 (I expect the problem will apply to both) - I got the Amber 18 source on Friday and have attempted to build `Amber-18-foss-2018b-AmberTools-18-patchlevel-10-8.eb` but it fails.

```
[snip a large amount of the log file]

Skipping bundled ARPACK build

mpicxx -o cpptraj ActionFrameCounter.o ActionList.o Action_Align.o Action_Angle.o Action_AreaPerMol.o Action_AtomMap.o Action_AtomicCorr.o Action_AtomicFluct.o Action_AutoImage.o Action_Average.o Action_Bounds.o Action_Box.o Action_Center.o Action_Channel.o Action_CheckChirality.o Action_CheckStructure.o Action_Closest.o Action_ClusterDihedral.o Action_Contacts.o Action_CreateCrd.o Action_CreateReservoir.o Action_DNAionTracker.o Action_DSSP.o Action_Density.o Action_Diffusion.o Action_Dihedral.o Action_Dipole.o Action_Distance.o Action_DistRmsd.o Action_Energy.o Action_FilterByData.o Action_FixAtomOrder.o Action_FixImagedBonds.o Action_GIST.o Action_Grid.o Action_GridFreeEnergy.o Action_HydrogenBond.o Action_Image.o Action_InfraredSpectrum.o Action_Jcoupling.o Action_LESsplit.o Action_LIE.o Action_LipidOrder.o Action_MakeStructure.o Action_Mask.o Action_Matrix.o Action_MinImage.o Action_Molsurf.o Action_MultiDihedral.o Action_MultiVector.o Action_NAstruct.o Action_NativeContacts.o Action_NMRrst.o Action_OrderParameter.o Action_Outtraj.o Action_PairDist.o Action_Pairwise.o Action_Principal.o Action_Projection.o Action_Pucker.o Action_Radgyr.o Action_Radial.o Action_RandomizeIons.o Action_Remap.o Action_ReplicateCell.o Action_Rmsd.o Action_Rotate.o Action_RunningAvg.o Action_STFC_Diffusion.o Action_Scale.o Action_SetVelocity.o Action_Spam.o Action_Strip.o Action_Surf.o Action_SymmetricRmsd.o Action_Temperature.o Action_Translate.o Action_Unstrip.o Action_Unwrap.o Action_Vector.o Action_VelocityAutoCorr.o Action_Volmap.o Action_Volume.o Action_Watershell.o AnalysisList.o Analysis_AmdBias.o Analysis_AutoCorr.o Analysis_Average.o Analysis_Clustering.o Analysis_Corr.o Analysis_ConstantPHStats.o Analysis_CrankShaft.o Analysis_CrdFluct.o Analysis_CrossCorr.o Analysis_CurveFit.o Analysis_Divergence.o Analysis_FFT.o Analysis_Hist.o Analysis_Integrate.o Analysis_IRED.o Analysis_KDE.o Analysis_Lifetime.o Analysis_LowestCurve.o Analysis_Matrix.o Analysis_MeltCurve.o Analysis_Modes.o Analysis_Multicurve.o Analysis_MultiHist.o Analysis_Overlap.o Analysis_PhiPsi.o Analysis_Regression.o Analysis_RemLog.o Analysis_Rms2d.o Analysis_RmsAvgCorr.o Analysis_Rotdif.o Analysis_RunningAvg.o Analysis_Spline.o Analysis_State.o Analysis_Statistics.o Analysis_TI.o Analysis_Timecorr.o Analysis_VectorMath.o Analysis_Wavelet.o ArgList.o Array1D.o AssociatedData.o Atom.o AtomMap.o AtomMask.o AtomType.o AtomTypeArray.o AxisType.o BondSearch.o Box.o BufferedFrame.o BufferedLine.o ByteRoutines.o CharMask.o CIFfile.o ClusterDist.o ClusterList.o ClusterMap.o ClusterMatrix.o ClusterNode.o ClusterSieve.o Cluster_DBSCAN.o Cluster_DPeaks.o Cluster_HierAgglo.o Cluster_Kmeans.o Cluster_ReadInfo.o Cmd.o CmdInput.o CmdList.o ComplexArray.o Constraints.o Control.o CoordinateInfo.o Corr.o Cph.o CpptrajFile.o CpptrajState.o CpptrajStdio.o CurveFit.o DataFile.o DataFileList.o DataIO.o DataIO_CCP4.o DataIO_CharmmFastRep.o DataIO_CharmmOutput.o DataIO_CharmmRepLog.o DataIO_CharmmRtfPrm.o DataIO_Cmatrix.o DataIO_Cpout.o DataIO_Evecs.o DataIO_Gnuplot.o DataIO_Grace.o DataIO_OpenDx.o DataIO_Mdout.o DataIO_NC_Cmatrix.o DataIO_RemLog.o DataIO_Std.o DataIO_VecTraj.o DataIO_Xplor.o DataIO_XVG.o DataSet.o DataSetList.o DataSet_1D.o DataSet_3D.o DataSet_Cmatrix.o DataSet_Cmatrix_DISK.o DataSet_Cmatrix_MEM.o DataSet_Cmatrix_NOMEM.o DataSet_Coords.o DataSet_Coords_CRD.o DataSet_Coords_REF.o DataSet_Coords_TRJ.o DataSet_GridDbl.o DataSet_GridFlt.o DataSet_Mat3x3.o DataSet_MatrixDbl.o DataSet_MatrixFlt.o DataSet_Mesh.o DataSet_Modes.o DataSet_Parameters.o DataSet_pH.o DataSet_PHREMD_Explicit.o DataSet_PHREMD_Implicit.o DataSet_RemLog.o DataSet_Topology.o DataSet_Vector.o DataSet_double.o DataSet_float.o DataSet_integer.o DataSet_string.o Deprecated.o DihedralSearch.o DispatchObject.o DistRoutines.o Energy.o EnsembleIn.o EnsembleIn_Multi.o EnsembleIn_Single.o EnsembleNavigator.o EnsembleOut.o EnsembleOut_Multi.o EnsembleOut_Single.o EnsembleOutList.o Ewald.o Ewald_ParticleMesh.o Ewald_Regular.o Exec_Analyze.o Exec_Calc.o Exec_Change.o Exec_ClusterMap.o Exec_CombineCoords.o Exec_Commands.o Exec_CompareTop.o Exec_CrdAction.o Exec_CrdOut.o Exec_CreateSet.o Exec_DataFile.o Exec_DataFilter.o Exec_DataSetCmd.o Exec_GenerateAmberRst.o Exec_Help.o Exec_LoadCrd.o Exec_LoadTraj.o Exec_ParmBox.o Exec_ParmSolvent.o Exec_ParmStrip.o Exec_ParmWrite.o Exec_PermuteDihedrals.o Exec_Precision.o Exec_PrintData.o Exec_ReadData.o Exec_ReadEnsembleData.o Exec_ReadInput.o Exec_RotateDihedral.o Exec_RunAnalysis.o Exec_ScaleDihedralK.o Exec_SequenceAlign.o Exec_SortEnsembleData.o Exec_System.o Exec_Top.o Exec_Traj.o Exec_UpdateParameters.o Exec_ViewRst.o FileIO_Bzip2.o FileIO_Gzip.o FileIO_Mpi.o FileIO_MpiShared.o FileIO_Std.o FileName.o FileTypes.o Frame.o GridAction.o HistBin.o Hungarian.o ImageRoutines.o InputTrajCommon.o KDE.o MapAtom.o MaskToken.o Matrix_3x3.o MetaData.o Mol.o Mol2File.o NameType.o NC_Cmatrix.o NC_Routines.o NetcdfFile.o OutputTrajCommon.o PDBfile.o PairList.o Parallel.o ParallelNetcdf.o ParmFile.o Parm_Amber.o Parm_CharmmPsf.o Parm_CIF.o Parm_Gromacs.o Parm_Mol2.o Parm_PDB.o Parm_SDF.o Parm_Tinker.o ProgressBar.o ProgressTimer.o PubFFT.o Random.o Range.o RPNcalc.o ReferenceAction.o RemdReservoirNC.o Residue.o SDFfile.o SimplexMin.o Spline.o StringRoutines.o StructureCheck.o StructureMapper.o SymmetricRmsdCalc.o TextFormat.o Timer.o TinkerFile.o TopInfo.o Topology.o TorsionRoutines.o Traj_AmberCoord.o Traj_AmberNetcdf.o Traj_AmberRestart.o Traj_AmberRestartNC.o Traj_Binpos.o Traj_CharmmCor.o Traj_CharmmDcd.o Traj_CharmmRestart.o Traj_CIF.o Traj_Conflib.o Traj_GmxTrX.o Traj_GmxXtc.o Traj_Gro.o Traj_Mol2File.o Traj_NcEnsemble.o Traj_PDBfile.o Traj_SDF.o Traj_SQM.o Traj_Tinker.o TrajectoryFile.o TrajectoryIO.o TrajFrameCounter.o TrajinList.o Trajin_Multi.o Trajin_Single.o TrajIOarray.o Trajout_Single.o TrajoutList.o VariableArray.o Vec3.o ViewRst.o Action_Esander.o Command.o Cpptraj.o Energy_Sander.o ReadLine.o main.o molsurf.o  pub_fft.o readline/libreadline.a -L/rds/bear-apps/devel/2018b/branfosj-eb-1/EL7/EL7-sandybridge/software/Amber/18-foss-2018b-AmberTools-18-patchlevel-10-8/lib64 -larpack -L/rds/bear-apps/devel/2018b/branfosj-eb-1/EL7/EL7-sandybridge/software/Amber/18-foss-2018b-AmberTools-18-patchlevel-10-8/lib64 -llapack -L/rds/bear-apps/devel/2018b/branfosj-eb-1/EL7/EL7-sandybridge/software/Amber/18-foss-2018b-AmberTools-18-patchlevel-10-8/lib64 -lblas -lgfortran -L/rds/bear-apps/devel/2018b/branfosj-eb-1/EL7/EL7-sandybridge/software/netCDF/4.6.1-foss-2018b/lib64 -lnetcdf -lbz2 -lz xdrfile/libxdrfile.a    
/rds/bear-apps/devel/2018b/branfosj-eb-1/EL7/EL7-sandybridge/software/binutils/2.30-GCCcore-7.3.0/bin/ld: cannot find -llapack
/rds/bear-apps/devel/2018b/branfosj-eb-1/EL7/EL7-sandybridge/software/binutils/2.30-GCCcore-7.3.0/bin/ld: cannot find -lblas
collect2: error: ld returned 1 exit status
make[4]: *** [cpptraj] Error 1
make[4]: Leaving directory `/rds/bear-apps/devel/2018b/branfosj-eb-1/EL7/EL7-sandybridge/software/Amber/18-foss-2018b-AmberTools-18-patchlevel-10-8/AmberTools/src/cpptraj/src'
make[3]: *** [install] Error 2
make[3]: Leaving directory `/rds/bear-apps/devel/2018b/branfosj-eb-1/EL7/EL7-sandybridge/software/Amber/18-foss-2018b-AmberTools-18-patchlevel-10-8/AmberTools/src/cpptraj'
make[2]: *** [build_cpptraj] Error 2
make[2]: Leaving directory `/rds/bear-apps/devel/2018b/branfosj-eb-1/EL7/EL7-sandybridge/software/Amber/18-foss-2018b-AmberTools-18-patchlevel-10-8/AmberTools/src'
make[1]: *** [serial] Error 2
make[1]: Leaving directory `/rds/bear-apps/devel/2018b/branfosj-eb-1/EL7/EL7-sandybridge/software/Amber/18-foss-2018b-AmberTools-18-patchlevel-10-8/AmberTools/src'
make: *** [install] Error 2
 (at easybuild/tools/run.py:501 in parse_cmd_output)
== 2018-11-26 13:34:06,864 easyblock.py:2855 WARNING build failed (first 300 chars): cmd " make install " exited with exit code 2 and output:
cd /rds/bear-apps/devel/2018b/branfosj-eb-1/EL7/EL7-sandybridge/software/Amber/18-foss-2018b-AmberTools-18-patchlevel-10-8/AmberTools/src && make install
make[1]: Entering directory `/rds/bear-apps/devel/2018b/branfosj-eb-1/EL7/EL7-sandybridge
== 2018-11-26 13:34:06,864 easyblock.py:284 INFO Closing log for application name Amber version 18
```

Let me know if there is any more information that I can provide to help here. Also, I will be attempting a fosscuda build at some point, but I have to build the complete stack for that before I can get to Amber.

**Comment by akesandgren** [comment]
Gah, one more thing that needs to be patched.
Will fix soon...

**Comment by akesandgren** [comment]
@branfosj Patch commited, can you check if it works? And if there is anything else you can find problems with...

**Comment by branfosj** [comment]
#### Test result
Build succeeded for 7 out of 7

#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _Boost-1.67.0-fosscuda-2018b.eb_ 
 * **SUCCESS** _netCDF-4.6.1-fosscuda-2018b.eb_ 
 * **SUCCESS** _Tkinter-2.7.15-fosscuda-2018b-Python-2.7.15.eb_ 
 * **SUCCESS** _Boost.Python-1.67.0-fosscuda-2018b-Python-2.7.15.eb_ 
 * **SUCCESS** _matplotlib-2.2.3-fosscuda-2018b-Python-2.7.15.eb_ 
 * **SUCCESS** _netCDF-Fortran-4.4.4-fosscuda-2018b.eb_ 
 * **SUCCESS** _Amber-18-fosscuda-2018b-AmberTools-18-patchlevel-10-8.eb_ 

https://gist.github.com/branfosj/fcd695a5859a62770ba20767e4ed0ad2

**Comment by zao** [comment]
Test report by @zao
**FAILED**
Build succeeded for 8 out of 9 (1 easyconfigs in this PR)
freja - Linux ubuntu 18.04, AMD Ryzen 7 1700 Eight-Core Processor, Python 2.7.15rc1
See https://gist.github.com/94c82f781986adcf1f79422ab29954a4 for a full test report.

**Comment by zao** [comment]
> Test report by @zao
> **FAILED**

No compatible GPU nor NV drivers in the machine, so rather expected.

**Comment by zao** [comment]
Build succeeded on the above machine, I unfortunately didn't run with test reports on for Amber itself.

**Comment by boegel** [comment]
@akesandgren How to proceed with this?

**Comment by akesandgren** [comment]
Well, it builds and works here. Do you want a fresh test report?
There's not much more I can do at the moment. As far as I can see I have addressed all questions/problems.

**Comment by akesandgren** [comment]
Updated cpptraj link patch to handle zlib/bzip2 the same way #6972 does.

This really needs a build report from someone else, @migueldiascosta ??

**Comment by branfosj** [comment]
#### Test result
Build succeeded for 7 out of 7

#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _Boost-1.67.0-fosscuda-2018b.eb_ 
 * **SUCCESS** _netCDF-4.6.1-fosscuda-2018b.eb_ 
 * **SUCCESS** _Tkinter-2.7.15-fosscuda-2018b-Python-2.7.15.eb_ 
 * **SUCCESS** _Boost.Python-1.67.0-fosscuda-2018b-Python-2.7.15.eb_ 
 * **SUCCESS** _matplotlib-2.2.3-fosscuda-2018b-Python-2.7.15.eb_ 
 * **SUCCESS** _netCDF-Fortran-4.4.4-fosscuda-2018b.eb_ 
 * **SUCCESS** _Amber-18-fosscuda-2018b-AmberTools-18-patchlevel-10-8.eb_ 

https://gist.github.com/branfosj/eb9ce78117c5a478dda46622970f7930

**Comment by akesandgren** [comment]
Kicking travis...

**Comment by migueldiascosta** [comment]
@boegel I still don't have access to Amber 18... :(

**Comment by boegel** [comment]
@migueldiascosta Do you think this is OK to merge based on @akesandgren and @branfosj's test reports?

**Comment by mboisson** [comment]
These patches fail for me with 
```
      pimd_vars.o pimd_force.o pimd_init.o cmd_vars.o cmd_matrix.o lscivr_vars.o lsc_init.o lsc_xp.o force.o -L/home/mboisson/.local/easybuild/software/2017/avx2/MPI/gcc5.4/cuda9.0/openmpi2.1/amber/18.10/lib -lsqm \
      nfe-abmd-hooks.o nfe-bbmd-ctxt.o nfe-bbmd-hooks.o nfe-colvar.o nfe-colvar-math.o nfe-colvar-type.o nfe-colvar-utils.o nfe-constants.o nfe-cv-ANGLE.o nfe-cv-COM_ANGLE.o nfe-cv-COM_DISTANCE.o nfe-cv-COM_TORSION.o nfe-cv-COS_OF_DIHEDRAL.o nfe-cv-DISTANCE.o nfe-cv-HANDEDNESS.o nfe-cv-LCOD.o nfe-cv-MULTI_RMSD.o nfe-cv-N_OF_BONDS.o nfe-cv-N_OF_STRUCTURES.o nfe- cv-R_OF_GYRATION.o nfe-cv-TORSION.o nfe-pmd-hooks.o nfe-rmsd.o nfe-sander-hooks.o nfe-sander-proxy.o nfe-stsm-hooks.o nfe-cv-PCA.o nfe-read-pca.o nfe-cv-SIN_OF_DIHEDRAL.o nfe-cv-PAIR_DIHEDRAL.o nfe-cv-PATTERN_DIHEDRAL.o nfe-cv-DF_COM_DISTANCE.o nfe-smd-hooks.o nfe-umbrella.o nfe-utils.o nfe-value.o mt19937.o xray_fftpack.o xray_FFT.o xray_fourier.o xray_globals.o xray_interface.o xray_real_space.o xray_reciprocal_space.o xray_utils.o \
      -lFpbsa ../lib/nxtsec.o /home/mboisson/.local/easybuild/software/2017/avx2/MPI/gcc5.4/cuda9.0/openmpi2.1/amber/18.10/lib/libemil.a -lstdc++ \
      ../sebomd/sebomd.a \
      ../lib/sys.a -L/cvmfs/soft.computecanada.ca/easybuild/software/2017/avx2/Compiler/gcc5.4/netcdf/4.4.1.1/lib -lnetcdff -lnetcdf \
      -lrism -lfftw3xf_intel -Wl,--start-group /cvmfs/soft.computecanada.ca/easybuild/software/2017/Core/imkl/11.3.4.258/mkl/lib/intel64/libmkl_intel_lp64.a /cvmfs/soft.computecanada.ca/easybuild/software/2017/Core/imkl/11.3.4.258/mkl/lib/intel64/libmkl_sequential.a /cvmfs/soft.computecanada.ca/easybuild/software/2017/Core/imkl/11.3.4.258/mkl/lib/intel64/       libmkl_core.a -Wl,--end-group -lpthread -ldl -larpack  -lxblas-amb -Wl,--start-group /cvmfs/soft.computecanada.ca/easybuild/software/2017/Core/imkl/11.3.4.258/mkl/lib/intel64/libmkl_intel_lp64.a /cvmfs/soft.computecanada.ca/easybuild/software/2017/Core/imkl/11.3.4.258/mkl/lib/intel64/libmkl_sequential.a /cvmfs/soft.computecanada.ca/easybuild/software/2017/Core/ imkl/11.3.4.258/mkl/lib/intel64/libmkl_core.a -Wl,--end-group -lpthread -ldl  \
      -fopenmp     Plumed.o
/cvmfs/soft.computecanada.ca/nix/store/xs99yskj0vhrf6cmn2048i870r8nzqm0-binutils-2.28/bin/ld: cannot find -lfftw3xf_intel
``` 

I believe the problem is that this patch assumes that MKL include the FFTW bindings and we don't have them ?    @bartoldeman would probably know for sure.

**Comment by mboisson** [comment]
I think the order of the selected FFTW in `Amber-18-AT-18_use_fftw_from_mkl_or_external.patch`   should be swapped. If `$EBROOTFFTW` is defined, it should use that one rather than attempt to use the one from MKL.

File: easybuild/easyconfigs/a/Amber/Amber-18-AT-18_use_fftw_from_mkl_or_external.patch [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+-    if [ $ncerror -gt 0 ]; then
+-        echo "Error: FFTW configure returned $ncerror"
+-        echo "       FFTW configure failed! Check the fftw3_config.log file"
+-        echo "       in the $AMBERHOME/AmberTools/src directory."
+-        exit 1
++    if [ -n "$MKL_HOME" ]; then
++	echo
++	echo "Using FFTW3 from MKL"
++	echo
++	flibs_fftw3="-lfftw3xf_intel $flibs_mkl"
```
Comment (mboisson) [inline easybuild/easyconfigs/a/Amber/Amber-18-AT-18_use_fftw_from_mkl_or_external.patch]: This line is an issue if one uses GCC + OpenMPI + MKL. I suggest swapping these two conditions (i.e. have `$EBROOTFFTW` take precedence over `$MKL_HOME`. If one has `FFTW` loaded, it should be used rather than MKL's.

Comment (akesandgren) [reply easybuild/easyconfigs/a/Amber/Amber-18-AT-18_use_fftw_from_mkl_or_external.patch]: Yeah, that makes sense. I'll change the patch and make some tests.

Comment (akesandgren) [reply easybuild/easyconfigs/a/Amber/Amber-18-AT-18_use_fftw_from_mkl_or_external.patch]: Have you tested if it works for you?

File: easybuild/easyconfigs/a/Amber/Amber-18-AT-18_use_fftw_from_mkl_or_external.patch [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
++	fftw3=""
++	fppflags="$fppflags -I$FFT_INC_DIR"
++    elif [ -n "$EBROOTFFTW" ]; then
++	echo
++	echo "Using external FFTW3"
++	echo
++	flibs_fftw3="-lfftw3"
++	if [ "$mpi" = 'yes' ]; then
++	    flibs_fftw3="-lfftw3_mpi $flibs_fftw3"
```
[the remark this answers is not in the document]
Comment (mboisson) [reply easybuild/easyconfigs/a/Amber/Amber-18-AT-18_use_fftw_from_mkl_or_external.patch]: This can probably be left this way. However, it does assume that one uses FFTW-MPI and not just FFTW.

**Comment by mboisson** [comment]
I think this is good to be merge @akesandgren @boegel. 

The only other comment I would have is that I personally don't like having  "AmberTools-18-patchlevel-10-8" in the version suffix. I have named my own recipe "AmberTools-18.10". Having the rest is too wordy for me.

**Comment by akesandgren** [comment]
Well, this is Amber, not just AmberTools. AmberTools has its own easyconfig (which I haven't made a ec for with these patches.

**Comment by mboisson** [comment]
Sorry, I mean't `Amber-18.10.11`, not `AmberTools`. 

Our recipe is this one : https://github.com/computecanada/easybuild-easyconfigs/tree/computecanada-master/easybuild/easyconfigs/a/Amber/Amber-18.10.11-gomklc-2016.4.11.90.eb

**Comment by akesandgren** [comment]
What about Amber18 with Ambertools 19?? how do you handle that?

**Comment by mboisson** [comment]
I probably would not handle that. Is there a need to handle it ? Aren't they usually version-locked together ?

**Comment by akesandgren** [comment]
No, Amber comes out bi-annually, AmberTools annually.
And users always wants the newest combo... last week

**Comment by mboisson** [comment]
We haven't run into this yet, but then I would split off Amber and AmberTools into separate modules.

**Comment by akesandgren** [comment]
Not sure you can do that due to the intricate dependencies between them...

**Comment by mboisson** [comment]
Mmm, then versioning it   `18.10.18.11`  might make sense ?

**Comment by mboisson** [comment]
I would personally skip the `-AT`, and put it in the description instead  (i.e. `18.10-18.11`). I like very clean and short versions and module avail/spider/list commands output.

**Comment by mboisson** [comment]
But either way, our easyconfig and yours don't have to be versioned exactly the same way, and it should not block the merge of this PR.

**Comment by akesandgren** [comment]
I just used the naming it originally had in EB when I started buidling it.
But Amber-x.y-AT-k.l feels better to me. MIght consider changing....
Or perhaps Amber-x.y-AmberTools-k.l so users don't get too confused.

**Comment by vanzod** [comment]
Test report by @vanzod
**FAILED**
Build succeeded for 6 out of 7 (7 easyconfigs in this PR)
cermis - Linux debian 9.6, Intel(R) Xeon(R) CPU           E5620  @ 2.40GHz, Python 2.7.13
See https://gist.github.com/68ea651740bcde4d68b68db9fea6bd2a for a full test report.

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 7 out of 7 (7 easyconfigs in this PR)
b-cn0315.hpc2n.umu.se - Linux ubuntu 16.04, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 2.7.12
See https://gist.github.com/20176846a6e649e9f571977025409e84 for a full test report.

**Comment by migueldiascosta** [comment]
Going in, thanks @akesandgren!
